### PR TITLE
Added connection to camera by hostname/IP

### PR DESCRIPTION
**Pull request by blaz-kranjc**
Added a feature where the connection to the camera can be established by providing hostname/IP as the device_id.

### CODE REVIEW

File: src/arvgvinterface.c [hunk 1]
```diff
@@ -405,6 +407,78 @@ arv_gv_interface_update_device_list (ArvInterface *interface, GArray *device_ids
 	}
 }
 
+static GInetAddress *
+arv_gv_interface_camera_locate(ArvGvInterface *gv_interface, GInetAddress *device_address) {
```
Comment (EmmanuelP) [inline src/arvgvinterface.c]: Please follow coding style: space before ( and { on the next line.

File: src/arvgvinterface.c [hunk 2]
```diff
@@ -405,6 +407,78 @@ arv_gv_interface_update_device_list (ArvInterface *interface, GArray *device_ids
...
+		return interface_address;
+	}
+
+	/* Just read a random register from the camera socket */
+	packet = arv_gvcp_packet_new_read_register_cmd(ARV_GVBS_N_STREAM_CHANNELS_OFFSET, 0, &size);
+	GSocketAddress *device_socket_address = g_inet_socket_address_new(device_address, ARV_GVCP_PORT);
+
+	for (iter = socket_list->sockets; iter != NULL; iter = iter->next) {
+		ArvGvDiscoverSocket *socket = iter->data;
+		GError *error = NULL;
```
Comment (EmmanuelP) [inline src/arvgvinterface.c]: coding style: blank line after var declaration

File: src/arvgvinterface.c [hunk 3]
```diff
@@ -424,8 +498,43 @@ _open_device (ArvInterface *interface, const char *device_id)
 	} else
 		device_infos = g_hash_table_lookup (gv_interface->priv->devices, device_id);
 
-	if (device_infos == NULL)
-		return NULL;
+	if (device_infos == NULL) {
+		// Try if device_id is a hostname/IP address
```
Comment (EmmanuelP) [inline src/arvgvinterface.c]: Use C style comments

File: src/arvgvinterface.c [hunk 4]
```diff
@@ -405,6 +407,78 @@ arv_gv_interface_update_device_list (ArvInterface *interface, GArray *device_ids
...
+	char buffer[ARV_GV_INTERFACE_SOCKET_BUFFER_SIZE];
+	GSList *iter;
+	size_t size;
+	int i, count;
+
+	socket_list = arv_gv_discover_socket_list_new ();
+
+	if (socket_list->n_sockets < 1) {
+		arv_gv_discover_socket_list_free (socket_list);
+		return interface_address;
```
Comment (EmmanuelP) [inline src/arvgvinterface.c]: return NULL;

File: src/arvgvinterface.c [hunk 5]
```diff
@@ -405,6 +407,78 @@ arv_gv_interface_update_device_list (ArvInterface *interface, GArray *device_ids
...
+	}
+
+	arv_gvcp_packet_free (packet);
+
+	do {
+		/* Now parse the result */
+		if (g_poll (socket_list->poll_fds, socket_list->n_sockets,
+					ARV_GV_INTERFACE_DISCOVERY_TIMEOUT_MS) == 0) {
+			arv_gv_discover_socket_list_free (socket_list);
+			return interface_address;
```
Comment (EmmanuelP) [inline src/arvgvinterface.c]: return NULL;

**Comment by EmmanuelP** [comment]
These changes make aravis crash:

./tests/arv-camera-test -n dummy

Thread 1 "lt-arv-camera-t" received signal SIGSEGV, Segmentation fault.
0x00007ffff5c562f2 in free () from /lib64/libc.so.6
Missing separate debuginfos, use: dnf debuginfo-install audit-libs-2.6.7-1.fc25.x86_64 glib2-2.50.1-1.fc25.x86_64 libblkid-2.28.2-1.fc25.x86_64 libcap-2.25-2.fc25.x86_64 libcap-ng-0.7.8-1.fc25.x86_64 libffi-3.1-9.fc24.x86_64 libgcc-6.2.1-2.fc25.x86_64 libmount-2.28.2-1.fc25.x86_64 libselinux-2.5-12.fc25.x86_64 libusbx-1.0.21-1.fc25.x86_64 libuuid-2.28.2-1.fc25.x86_64 libxml2-2.9.3-4.fc25.x86_64 nss-mdns-0.10-17.fc24.x86_64 pcre-8.39-6.fc25.x86_64 systemd-container-231-10.fc25.x86_64 systemd-libs-231-10.fc25.x86_64 xz-libs-5.2.2-2.fc24.x86_64 zlib-1.2.8-10.fc24.x86_64
(gdb) bt full
#0  0x00007ffff5c562f2 in free () at /lib64/libc.so.6
#1  0x00007ffff5cbfecd in freeaddrinfo () at /lib64/libc.so.6
#2  0x00007ffff7bac8dd in _open_device (interface=interface@entry=0x611470, device_id=device_id@entry=0x610390 "dummy") at arvgvinterface.c:515
        hints = 
          {ai_flags = 0, ai_family = 2, ai_socktype = 1, ai_protocol = 0, ai_addrlen = 0, ai_addr = 0x0, ai_canonname = 0x0, ai_next = 0x0}
        servinfo = 0x7ffff72beb15 <g_type_register_static_simple+181>
        endpoint = <optimized out>
        gv_interface = <optimized out>
        device = 0x0
        device_infos = 0x0
        device_address = <optimized out>
#3  0x00007ffff7bac90f in arv_gv_interface_open_device (interface=0x611470, device_id=0x610390 "dummy") at arvgvinterface.c:557
        device = <optimized out>
#4  0x00007ffff7b9112e in arv_open_device (device_id=0x610390 "dummy")
    at arvsystem.c:241
        interface = <optimized out>
        device = 0x0
        i = <optimized out>
#5  0x00007ffff7b9a306 in arv_camera_new (name=<optimized out>)
    at arvcamera.c:1945
---Type <return> to continue, or q <return> to quit---
        device = <optimized out>
#6  0x0000000000401eed in main (argc=<optimized out>, argv=<optimized out>) at arvcameratest.c:251
        data = {main_loop = 0x1, buffer_count = 0, chunk_parser = 0x0, chunks = 0x0}
        camera = <optimized out>
        stream = <optimized out>
        context = 0x60fdd0
        error = 0x0
        i = <optimized out>

**Comment by blaz-kranjc** [comment]
Sorry about that.

**Comment by EmmanuelP** [comment]
Pushed to master. Thanks a lot.

**Pull request closed without merging**